File: matlab/task1_4.m
% 1.4. Controle de brilho aditivo
% (valor do pixel resultante = valor do pixel original + c, c inteiro)
function task1_4()
    assetsDir = ['..' filesep 'assets' filesep];
    filePath = strcat(assetsDir, 'lenna.png');
    
    BRIGHTNESS_ADJUSTMENT = 60;
    
    originalImage = imread(filePath);
    modifiedImage = AdditiveBrightnessControl(originalImage, BRIGHTNESS_ADJUSTMENT);
      
    % Ploting
    fig=figure();        
    subplot(1,3,1.3), imshow(originalImage,[]), title('Original image')
    subplot(1,3,2.5), imshow(modifiedImage,[]), title('Modified image')
       
    grayscaleImage  = rgb2gray(originalImage);
    modifiedGrayscaleImage = AdditiveBrightnessControl(grayscaleImage, BRIGHTNESS_ADJUSTMENT);
    
    fig2=figure();        
    subplot(1,3,1.3), imshow(grayscaleImage,[]), title('Original Grayscale')
    subplot(1,3,2.5), imshow(modifiedGrayscaleImage,[]), title('Modified Grayscale')
    
    saveas(fig,strcat('output',filesep,mfilename,'_1.png'));
    saveas(fig2,strcat('output',filesep,mfilename,'_2.png'));
end

function output = AdditiveBrightnessControl(input, brightness)
    % MatLab shortcut
    % The pixel type uint8 cannot go higher than 255.
    output = uint8(brightness + input);
end

File: python/task1_4.py
# coding=UTF-8
# 1.4. Controle de brilho aditivo
# (valor do pixel resultante = valor do pixel original + c, c inteiro)

import utils

BRIGHTNESS_ADJUSTMENT = 80


def main():
    image = utils.load_image('lenna.png')

    modified_image = image + BRIGHTNESS_ADJUSTMENT
    utils.fit_matrix_in_interval(modified_image)

    utils.display_single_image('Original', image)
    utils.display_single_image('Modified', modified_image)

    utils.wait_key_and_destroy_windows()


if __name__ == "__main__":
    main()